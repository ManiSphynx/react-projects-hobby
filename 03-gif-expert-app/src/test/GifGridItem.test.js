import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../components/GifGridItem";

describe("Pruebas en GrifGridItem.js", () => {
  const title = "Un titulo";
  const url = "https://www.googleapis.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual url y alt de los props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");

    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
