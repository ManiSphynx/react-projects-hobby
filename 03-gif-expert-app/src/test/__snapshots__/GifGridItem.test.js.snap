// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GrifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="Un titulo"
    src="https://www.googleapis.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
